src/borg/helpers.py

```diff
@@ -1696,3 +1696,26 @@ def write(self, s):
                 except OSError:
                     pass
         return len(s)
+
+
+def swidth_slice(string, desired_length):
+    """
+    Return a slice of *desired_length* from *string*.
+
+    Negative *desired_length* means from the end of string.
```

Comment:
some minor wording issues in param names and docstring:
- length vs. width, usually they are orthogonal
- it should mention "desired width when output to a terminal" or "desired screen width" - if one does not mention terminal or screen, it looks strange why we need such a complicated function for that.
- it could also mention CJK glyphs as an example for needing more that 1 terminal cell width